docs/using-keystone-webhook-authenticator-and-authorizer.md

```diff
@@ -334,53 +410,111 @@ and both as well, depending on your requirement.
-  ```
-
-- Modify API server config file to use the webhook service for authorization.
-
-  ```shell
-  sed -i '/image:/ i \ \ \ \ - --authorization-webhook-config-file=/etc/kubernetes/pki/webhookconfig.yaml' /etc/kubernetes/manifests/kube-apiserver.yaml
-  sed -i "/authorization-mode/c \ \ \ \ - --authorization-mode=Node,Webhook,RBAC" /etc/kubernetes/manifests/kube-apiserver.yaml
-  ```
-
-- Wait for the API server to restart successfully until you can get all the
-  pods in `kube-system` namespace by running `kubectl get pod -n kube-system`
+- Create webhook config file. We reuse the folder `/etc/kubernetes/pki/`
+  because it's already mounted and accessible by API server pod.
+
+    ```shell
+    cat <<EOF > /etc/kubernetes/pki/webhookconfig.yaml
+    ---
+    apiVersion: v1
+    kind: Config
+    preferences: {}
+    clusters:
+      - cluster:
+          insecure-skip-tls-verify: true
+          server: https://${keystone_auth_service_addr}:8443/webhook
+        name: webhook
+    users:
+      - name: webhook
+    contexts:
+      - context:
+          cluster: webhook
+          user: webhook
+        name: webhook
+    current-context: webhook
+    ```
+
+- Modify kube-apiserver config file to use the webhook service for
+  authentication and authorization.
+
+    ```shell
+    sed -i '/image:/ i \ \ \ \ - --authentication-token-webhook-config-file=/etc/kubernetes/pki/webhookconfig.yaml' /etc/kubernetes/manifests/kube-apiserver.yaml
+    sed -i '/image:/ i \ \ \ \ - --authorization-webhook-config-file=/etc/kubernetes/pki/webhookconfig.yaml' /etc/kubernetes/manifests/kube-apiserver.yaml
+    sed -i "/authorization-mode/c \ \ \ \ - --authorization-mode=Node,Webhook,RBAC" /etc/kubernetes/manifests/kube-apiserver.yaml
+    ```
+
+- Wait for the API server to restart successfully until you can see all the
+  pods are running in `kube-system` namespace.
+
+## Authorization policy definition(version 2)
+
+The version 2 definition could be used together with version 1 but will
+take precedence over version 1 if both are defined. The version 1
+definition is still supported but is considered deprecated.
+
+The authorization policy definition is based on whitelist, which means
+the operation is allowed if *ANY* rule defined in the permissions is
+satisfied.
+
+- "users" defines which projects the OpenStack users belong to and what
+  roles they have.
+- "resource_permissions" is a map with the key defines namespaces and
+  resources, the value defines the allowed operations. `!` is supported
+  both for namespaces and resources, see examples below.
+- "nonresource_permissions" is a map with the key defines the
+  non-resource endpoint such as `/healthz`, the value defines the
+  allowed operations.
+
+Some examples:
+
+- Any operation is allowed on any resource in any namespace.
+
+    ```json
+    "resource_permissions": {
+      "*.*": ["*"]
```

Comment:
So `*.*` here is mainly used to set rules on operating resources of a certain namespace?
We could use other separators, e.g. `/`, instead of  `.`, which is misleading and has special meanings in regex.

Reply:
Actually I'd prefer to have `resources` and `namespaces` set separately.

e.g.
```json
"resource_permissions": [
  {
    "namespaces": "foo*",
    "resources": "bar,baz",
    "verbs": "get,list,delete"
  }
]
```

Reply:
I intentionally avoided using the keys such as namespaces, resources and verbs, because in this way, there is not much difference with version one which is tedious and redundant.

Reply:
> So `*.*` here is mainly used to set rules on operating resources of a certain namespace?
> We could use other separators, e.g. `/`, instead of `.`, which is misleading and has special meanings in regex.

Make sense, will consider.